dll/win32/browseui/explorerband.cpp

```diff
@@ -261,6 +262,50 @@ CExplorerBand::NodeInfo* CExplorerBand::GetNodeInfo(HTREEITEM hItem)
     return reinterpret_cast<NodeInfo*>(tvItem.lParam);
 }
 
+static HRESULT GetCurrentLocationFromView(IShellView &View, PIDLIST_ABSOLUTE &pidl)
+{
+    CComPtr<IFolderView> pfv;
+    CComPtr<IShellFolder> psf;
+    HRESULT hr = View.QueryInterface(IID_PPV_ARG(IFolderView, &pfv));
+    if (SUCCEEDED(hr) && SUCCEEDED(hr = pfv->GetFolder(IID_PPV_ARG(IShellFolder, &psf))))
+        hr = SHELL_GetIDListFromObject(psf, &pidl);
+    return hr;
+}
+
+HRESULT CExplorerBand::GetCurrentLocation(PIDLIST_ABSOLUTE &pidl)
+{
+    pidl = NULL;
+    CComPtr<IShellBrowser> psb;
+    HRESULT hr = IUnknown_QueryService(m_pSite, SID_STopLevelBrowser, IID_PPV_ARG(IShellBrowser, &psb));
+    if (FAILED_UNEXPECTEDLY(hr))
+        return hr;
+
+    CComPtr<IBrowserService> pbs;
+    if (SUCCEEDED(hr = psb->QueryInterface(IID_PPV_ARG(IBrowserService, &pbs))))
+        if (SUCCEEDED(hr = pbs->GetPidl(&pidl)) && pidl)
+            return hr;
+
+    CComPtr<IShellView> psv;
+    if (!FAILED_UNEXPECTEDLY(hr = psb->QueryActiveShellView(&psv)))
+        if (SUCCEEDED(hr = psv.p ? GetCurrentLocationFromView(*psv.p, pidl) : E_FAIL))
+            return hr;
+    return hr;
+}
+
+HRESULT CExplorerBand::IsCurrentLocation(PCIDLIST_ABSOLUTE pidl)
+{
+    if (!pidl)
+        return E_INVALIDARG;
+    HRESULT hr = E_FAIL;
+    PIDLIST_ABSOLUTE location = m_pidlCurrent, free = NULL;
+    if (!location && SUCCEEDED(hr = GetCurrentLocation(location)))
+        free = location;
```

Comment:
Could do without the extra `free` thing by checking `if (m_pidlCurrent != location)`, but I have no strong preference for that style.

Reply:
Ideally m_pidlCurrent should always be valid when the tree is navigated somewhere. I don't know why it's allowed to be NULL but I don't want to deal with that in this PR.

Reply:
Fair enough.